src/HttpClientRequestAdapter.cs

```diff
@@ -388,7 +387,7 @@ private async Task ThrowIfFailedResponse(HttpResponseMessage response, Dictionar
 
             var statusCodeAsInt = (int)response.StatusCode;
             var statusCodeAsString = statusCodeAsInt.ToString();
-            var responseHeadersDictionary = response.Headers.ToDictionary(x => x.Key, y => y.Value, StringComparer.OrdinalIgnoreCase);
+            var responseHeadersDictionary = new Dictionary<string, IEnumerable<string>>(StringComparer.OrdinalIgnoreCase);

```

Comment:
The dictionary is no longer pre-populated with the response headers. Yes?

Reply:
late to the party: why not use the copy constructor instead of looping ourselves?
https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.-ctor?view=net-8.0#system-collections-generic-dictionary-2-ctor(system-collections-generic-idictionary((-0-1))-system-collections-generic-iequalitycomparer((-0)))